src/nvim/regexp_nfa.c

```diff
@@ -6019,6 +6031,7 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
   log_fd = NULL;
 #endif
 
+theend:
```

Comment:
Looks like it was used exclusively for OOM errors, which we obviate by using `xmalloc` from `memory.c`.